### PR TITLE
Feat: add Tooltip component

### LINKED ISSUE #22: Create Tooltip component
## Description
https://www.figma.com/file/nPJGejQqgLQfkVMJqhcCeO/Maquettes?node-id=138%3A797

### PR DESCRIPTION
**Pull request by c1-ra**
## Linked issues
Closes #22

### CODE REVIEW

**Comment by Damien-MATHIEU** [comment]
Very good! Some suggestions about the namings: 
- The file `Tooltip.typings.ts` should be rename `Tooltip.types.ts`
- The Scss file should not start with a capital letter, it should be named `tooltip.module.scss`

File: src/components/Tooltip/internal/Tooltip.module.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+@use '../../../style/mixins/font-config.scss';
+
+.tooltip {
+  color: map.get(colors.$colors, 'base-white');
+  position: relative;
+  &:hover span {
+    visibility: visible;
+  }
+  span {
+    visibility: hidden;
```
Comment (Damien-MATHIEU) [inline src/components/Tooltip/internal/Tooltip.module.scss:13]: On the tooltip we can add an animation so the text appearing is smoother

```suggestion
  &:hover span {
    transition: visibility 150ms ease, opacity 150ms ease;
    visibility: visible;
    opacity: 1;
  }
  span {
    transition: visibility 150ms ease, opacity 150ms ease;
    visibility: hidden;
    opacity: 0;
```

We cannot animate `visibility` alone so adding an opacity will help us do it

File: src/components/Tooltip/internal/Tooltip.module.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    background-color: map.get(colors.$colors, 'base-white');
+    color: map.get(colors.$colors, 'grey-dark');
+    position: absolute;
+    bottom: 100%;
+    border-radius: map.get(spaces.$spaces, '2');
+    margin: map.get(spaces.$spaces, '2');
+    padding: map.get(spaces.$spaces, '2');
+    margin-left: -1px;
```
Comment (Damien-MATHIEU) [inline src/components/Tooltip/internal/Tooltip.module.scss:21]: Doing that make the end of the tooltip tail align to the icon ~~for some reason~~.
```suggestion
    margin-left: 4px;
```

File: src/components/Tooltip/internal/Tooltip.module.scss (hunk 1 above)
Comment (c1-ra) [reply src/components/Tooltip/internal/Tooltip.module.scss]: It's better indeed!

File: src/components/Tooltip/internal/Tooltip.module.scss (hunk 2 above)
Comment (c1-ra) [reply src/components/Tooltip/internal/Tooltip.module.scss]: Thank you :)

**Review by Damien-MATHIEU — DISMISSED** [review]

**Review by moreirathomas — APPROVED** [review]

**Pull request merged**